packages/api-client-core/src/GadgetRecord.ts

```diff
@@ -29,7 +30,7 @@ export class GadgetRecord_<Shape extends RecordShape> {
 
   private empty = false;
 
-  constructor(data: Shape) {
+  constructor(data: Shape, readonly modelManager?: AnyModelManager) {
```

Comment:
I think it'd be prudent to store this under a symbol so that it can't collide with other stuff, like we do for the other props